libnetwork/osl/interface_linux.go

```diff
@@ -14,6 +14,28 @@ import (
 	"github.com/vishvananda/netns"
 )
 
+// newInterface creates a new interface in the given namespace using the
+// provided options.
+func newInterface(ns *Namespace, srcName, dstPrefix string, options ...IfaceOption) (*Interface, error) {
+	i := &Interface{
+		srcName: srcName,
+		dstName: dstPrefix,
+		ns:      ns,
+	}
+	for _, opt := range options {
+		if opt != nil {
+			opt(i)
```

Comment:
`IfaceOption` are only setters, so I think we can remove its `error` return type.

Reply:
OH! Actually, I think I messed up a rebase here (looks like I dropped the error return?)

I think we can consider the functional arguments to return an error so that an invalid option would actually produce an error (where it makes sense). In either case, I should move the code "as is" from the old location.

Reply:
> I should move the code "as is" from the old location.

Yeah that's probably better.

> I think we can consider the functional arguments to return an error so that an invalid option would actually produce an error (where it makes sense).

Except no `IfaceOption` actually return an error, as they're all data setters. So, I think we should simply change the declaration of `IfaceOption` to `func(i *Interface)` in a follow-up PR.

Reply:
I need to dig  in memory; ISTR I found some places where they _were_ setters, but other code then validated it _after_ it was set (maybe that was the `master` i this case (see the code below this), but there may have been other cases).

In either case; let's leave that for a follow-up.